dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/runner/WorkerTaskExecuteRunnable.java

```diff
@@ -325,6 +314,30 @@ protected void clearTaskExecPathIfNeeded() {
         }
     }
 
+    protected void writePodLodIfNeeded() {
+        if (null == taskExecutionContext.getK8sTaskExecutionContext()) {
+            return;
+        }
+        log.info("The current task is k8s task, begin to write pod log");
+        ProcessUtils.getPodLog(taskExecutionContext.getK8sTaskExecutionContext(), taskExecutionContext.getTaskAppId());
+    }
+
+    protected void closeLogAppender() {
+        try {
+            writePodLodIfNeeded();
+            if (RemoteLogUtils.isRemoteLoggingEnable()) {
+                RemoteLogUtils.sendRemoteLog(taskExecutionContext.getLogPath());
+                log.info("Log handler sends task log {} to remote storage asynchronously.",
+                        taskExecutionContext.getLogPath());
+            }
+        } catch (Exception ex) {
+            log.error("Write k8s pod log failed", ex);
+        } finally {
+            log.info(FINALIZE_SESSION_MARKER, FINALIZE_SESSION_MARKER.toString());
+
+        }
+    }
```

Comment:
It should be implemented in k8s task.

Reply:
> It should be implemented in k8s task.

Hi, @caishunfeng, spark on k8s also have to do this, collect pod log here looks good to me.

Reply:
If send remote log in `closeLogAppender`, we don't have to send it again in `afterExecute` and `afterThrowing`.

Reply:
If task success the `afterExecute` method will be executed, otherwise `afterThrowing` will be executed. So the log will not be send twice.

Reply:
Hi, @ruanwenjun , i mean send twice in `closeLogAppender` and `afterThrowing`/`afterExecute`.

Reply:
Good catch.